### DISCUSSION TITLE
Disable Renovate Gradle Artifact Updating?

### DISCUSSION THREAD

**Discussion by AlexanderGH** [renovatebot/renovate#31324] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket, I don't know the exact version, but one from August 2024

### Please tell us more about your question or problem

In our repo, we already handle artifact updating once PRs are send and our setup is incompatible with the built-in artifact generation/verification in renovate.

Is there some way to disable the following from running on dep version update PRs?
https://github.com/renovatebot/renovate/blob/62d88c6792615049d0dc3731d006501363f55d29/lib/modules/manager/gradle/artifacts.ts#L134

Ideally something that doesn't rely on renaming our lockfiles to something renovate doesn't detect.

**Comment by Churro** [comment]
A lean option could be to extend the [updateLockFiles](https://docs.renovatebot.com/configuration-options/#updatelockfiles) configuration option for gradle. Unless there's protest from one of the maintainers, I could come up with a PR for that.

**Reply by rarkins** [reply]
I also had the idea of a post upgrade task to reset the lock file back to origin SHA or something like that

**Reply by AlexanderGH** [reply]
One thing to clarify (i kept it vague because i wanted to avoid solutions specific to our setup), in our setup, we're actually unable to run that command at all so would prefer a solution that disables running it rather than running another command to reset its state. Some ideas i had:
- same as https://github.com/renovatebot/renovate/discussions/31324#discussioncomment-10611311 (add support for gradle to that option and apply it to dep update PRs too to allow disabling all lockfile updates)
- allow configuring the lockfile pattern so we can put in some pattern that never matches and no-ops that artifact method using the existing code
